genesis/engine/solvers/rigid/rigid_solver_decomp.py

```diff
@@ -2447,12 +2865,36 @@ def n_qs(self):
             return self._n_qs
         return sum([entity.n_qs for entity in self._entities])
 
+    @property
+    def max_n_qs_per_link(self):
+        if self.is_built:
+            return self._max_n_qs_per_link
+        return max(link.n_qs for link in self.links) if self.links else 0
+
     @property
     def n_dofs(self):
         if self.is_built:
             return self._n_dofs
         return sum(entity.n_dofs for entity in self._entities)
 
+    @property
+    def max_n_dofs_per_entity(self):
+        if self.is_built:
+            return self._max_n_dofs_per_entity
+        return max(entity.n_dofs for entity in self._entities) if self._entities else 0
+
+    @property
+    def max_n_dofs_per_link(self):
+        if self.is_built:
+            return self._max_n_dofs_per_link
+        return max(link.n_dofs for link in self.links) if self.links else 0
+
+    @property
+    def max_n_dofs_per_joint(self):
+        if self.is_built:
+            return self._max_n_dofs_per_joint
+        return max(joint.n_dofs for joint in self.joints) if self.joints else 0
+
```

Comment:
Do not expose additional function in the API if they are only used once at init, and I meaningless for users.